sphinx/util/inspect.py

```diff
@@ -158,3 +158,26 @@ def is_builtin_class_method(obj, attr_name):
     if not hasattr(builtins, safe_getattr(cls, '__name__', '')):
         return False
     return getattr(builtins, safe_getattr(cls, '__name__', '')) is cls
+
+
+def dumb_signature(functional_object):
+    src = inspect.getsourcelines(functional_object)[0]
```

Comment:
I have general concerns about this approach:  There are several shortcomings in this hand-rolled parser (see examples below), which might lead to function signatures being *completely* misparsed.  Reimplementing a Python parser with regular expressions is a poor idea anyways;  `ast.parse` might be a much better idea.

I'd rather tune `sphinx.ext.autodoc.formatargspec` to fetch names instead of values when this setting is enabled.  (I'd introduce a new parameter to this function which is fed from the config and then goes and tries to extract names instead of values from the source, failing gracefully if parsing fails for whatever reason.)

Also:  `inspect.getsourcelines` can fail with an `IOError` for non-source distributions (eg. `.pyc`).